src/adserver.js

```diff
@@ -0,0 +1,57 @@
+import {formatQS} from './url';
+
+//Adserver parent class
+var adserver = function(attr) {
+  var that = {};
+  that.name = attr.adserver;
+  that.code = attr.code;
+  that.getWinningBidByCode = function() {
+    var bidObject = $$PREBID_GLOBAL$$._bidsReceived.find(bid => bid.adUnitCode === that.code);
+    return bidObject;
+  };
+  return that;
+};
+
+//DFP ad server
+exports.dfpAdserver = function (options, urlComponents) {
+  var that = adserver(options);
+  that.urlComponents = urlComponents;
+
+  var dfpReqParams = {
+    'env' : 'vp',
+    'gdfp_req' : '1',
+    'impl' : 's',
+    'unviewed_position_start' : '1'
+  };
+
+  var dfpParamsWithVariableValue = ['output', 'iu', 'sz', 'url', 'correlator', 'description_url', 'hl'];
+
+  var getCustomParams = function(targeting) {
+    targeting.hb_pb = '10.00';
```

Comment:
assume this  a placeholder. We should expose an export on `dfpAdserver` that sets arbitrary key/value pairs. Then have the auction request that method.